src/Core/Localization/Specification/Factory.php

```diff
@@ -0,0 +1,277 @@
+        }
+
+        return $symbolsLists;
+    }
+
+    /**
+     * Get a NumberSymbolList object from a CLDR NumberSymbolsData object.
+     *
+     * @param NumberSymbolsData $symbolsData
+     *                                       Data that will be used to build the NumberSymbolList object
+     *
+     * @return NumberSymbolList
+     *                          An immutable NumberSymbolList object
+     *
+     * @throws LocalizationException
+     *                               If passed data is invalid
+     */
+    protected function getNumberSymbolList(NumberSymbolsData $symbolsData)
+    {
+        return new NumberSymbolList(
+            $symbolsData->getDecimal(),
+            $symbolsData->getGroup(),
+            $symbolsData->getList(),
+            $symbolsData->getPercentSign(),
+            $symbolsData->getMinusSign(),
+            $symbolsData->getPlusSign(),
+            $symbolsData->getExponential(),
+            $symbolsData->getSuperscriptingExponent(),
+            $symbolsData->getPerMille(),
+            $symbolsData->getInfinity(),
+            $symbolsData->getNan()
+        );
+    }
+
+    /**
+     * Extract the min number of fraction digits from a number pattern (decimal, currency, percentage).
+     *
+     * @param string $pattern
+     *                        The formatting pattern to use for extraction
+     *
+     * @return int
+     *             The min number of fraction digits to display in the final number
+     */
+    protected function getMinFractionDigits($pattern)
+    {
+        $dotPos = (int) strpos($pattern, '.');
+
+        return substr_count($pattern, '0', $dotPos);
+    }
+
+    /**
+     * Get the primary digits group size from a number formatting pattern.
+     *
+     * @param string $pattern
+     *                        The CLDR number formatting pattern (e.g.: #,##0.###)
+     *
+     * @return int
+     *             The primary group size of the passed pattern
+     */
+    protected function getPrimaryGroupSize($pattern)
+    {
+        $groups = $this->getPatternGroups($pattern);
+        $nbGroups = count($groups);
+
+        return strlen($groups[$nbGroups - 1]);
+    }
+
+    /**
+     * Get the secondary digits group size from a number formatting pattern
+     * e.g.: with    #,##0.### => No secondary group. Will return primary group size.
+     * e.g.: with #,##,##0.### => Secondary group size is 2, primary group size is 3.
+     *
+     * @param string $pattern
+     *                        The CLDR number formatting pattern
+     *
+     * @return int
+     *             The secondary group size of the passed pattern
+     */
+    protected function getSecondaryGroupSize($pattern)
+    {
+        $groups = $this->getPatternGroups($pattern);
+        $nbGroups = count($groups);
+
+        if ($nbGroups > 2) {
+            return strlen($groups[$nbGroups - 2]);
+        }
+
+        return strlen($groups[$nbGroups - 1]);
```

Comment:
```php
return strlen($groups[$nbGroups - (count($groups) > 2 ? 2 : 1)]);
```

Reply:
less readable

Reply:
Tsssss, you noob!

Reply:
I agree with @tomlev 
I'm all for ternary notations if it reduces efficiently the size of the code
But only if it's human readable, otherwise you lose too much time (and brain..) to read/understand it
Let's not underestimate mental load!